### PR TITLE
editoast: refactor infra tests part 5

### PR DESCRIPTION
**Pull request by hamz2a**
Part of #6980

`create_mini_infra` is a temporary infra, and will be replaced by `small_infra `once `Infra::persist` is refactored.

### CODE REVIEW

**Comment by leovalais** [comment]
Thanks for this PR. Can we please delay merging it until the OpenApi migration is complete? Sadly there's some overlap that I won't be able to fix since I'll be in vacation, and I'd like these to be merged asap :/

Cf. #7553 #7677 #7686

**Review by woshilapin — APPROVED** [review]
🚀

File: editoast/src/views/infra/auto_fixes/mod.rs [hunk 1]
```diff
@@ -380,100 +365,82 @@ mod tests {
...
         assert!(operations.contains(&Operation::Delete(DeleteOperation {
-            obj_id: "DA0".to_string(),
+            obj_id: "D1".to_string(),
             obj_type: ObjectType::Detector,
         })));
         assert!(operations.contains(&Operation::Delete(DeleteOperation {
-            obj_id: "PA0".to_string(),
+            obj_id: "switch".to_string(),
             obj_type: ObjectType::Switch,
         })));
```
Comment (leovalais) [inline editoast/src/views/infra/auto_fixes/mod.rs]: Better to assert over the whole collection (`assert_eq!(operations, ...)`): we want this test to fail if there is another error.

Another (weaker) way of doing so would be to check that there is no error before removing the track.

File: editoast/src/views/infra/objects.rs [hunk 2]
```diff
@@ -104,74 +110,76 @@ mod tests {
...
+        let db_pool = app.db_pool();
+        let empty_infra = create_empty_infra(db_pool.get_ok().deref_mut()).await;
 
         let req = TestRequest::post()
-            .uri(format!("/infra/{}/objects/TrackSection", empty_infra.id()).as_str())
+            .uri(format!("/infra/{}/objects/TrackSection", empty_infra.id).as_str())
             .set_json(["invalid_id"])
             .to_request();
-        let response = call_service(&app, req).await;
+        let response = call_service(&app.service, req).await;
```
Comment (leovalais) [inline editoast/src/views/infra/objects.rs]: Better use `app.fetch` when possible for consistency.

**Review by leovalais — CHANGES_REQUESTED** [review]
Thanks for letting all OpenApi PRs being merged before this one, it really helped! 😄 

Just a few minor details and we're good to go 🚀

File: editoast/src/views/infra/objects.rs (hunk 2 above)
Comment (hamz2a) [reply editoast/src/views/infra/objects.rs]: Done.

File: editoast/src/views/infra/objects.rs [hunk 3]
```diff
@@ -104,74 +110,76 @@ mod tests {
...
+        let empty_infra = create_empty_infra(db_pool.get_ok().deref_mut()).await;
 
         let req = TestRequest::post()
-            .uri(format!("/infra/{}/objects/TrackSection", empty_infra.id()).as_str())
+            .uri(format!("/infra/{}/objects/TrackSection", empty_infra.id).as_str())
             .set_json(vec![""; 0])
             .to_request();
-        assert_eq!(call_service(&app, req).await.status(), StatusCode::OK);
+        assert_eq!(
+            call_service(&app.service, req).await.status(),
```
Comment (leovalais) [inline editoast/src/views/infra/objects.rs]: Likewise about `app.fetch` for all other `call_service` in this file.

File: editoast/src/views/infra/routes.rs
Comment (leovalais) [inline editoast/src/views/infra/routes.rs:1]: Likewise about `app.fetch` for all other `call_service` in this file.

**Review by leovalais — APPROVED** [review]
Thanks for re-arranging the PR to avoid merging short-lived code.
A few more changes and we're good :)

File: editoast/src/views/infra/objects.rs (hunk 3 above)
Comment (hamz2a) [reply editoast/src/views/infra/objects.rs]: Done.

File: editoast/src/views/infra/routes.rs
Comment (hamz2a) [reply editoast/src/views/infra/routes.rs:1]: Done.

**Pull request merged**